packages/cli/src/credentials/credentials.controller.ts

```diff
@@ -242,6 +243,13 @@ export class CredentialsController {
 			data: preparedCredentialData.data as unknown as ICredentialDataDecryptedObject,
 		});
 
+		// Update isGlobal if provided in the payload and user has permission
+		const isGlobal = body.isGlobal;
+		const canShareGlobally = hasGlobalScope(req.user, 'credential:shareGlobally');
+		if (isGlobal !== undefined && canShareGlobally) {
```

Comment:
Wouldn't a better model be that this fails with forbidden error if user does not have the permission to share globally instead of failing silently?

Reply:
I was trying to simplify the UI so it can always pass the flag when updating. But could change this.